flow/model/qvalue/qvalue.go

```diff
@@ -21,157 +21,589 @@ import (
 )
 
 // if new types are added, register them in gob - cdc_records_storage.go
-type QValue struct {
-	Kind  QValueKind
-	Value interface{}
+type QValue interface {
+	Kind() QValueKind
+	Value() any
 }
 
-func (q QValue) Equals(other QValue) bool {
-	if q.Kind == QValueKindJSON {
-		return true // TODO fix
-	} else if q.Value == nil && other.Value == nil {
+type QValueNull QValueKind
```

Comment:
could be split up more, maybe a file for numeric types and helpers, time types and helpers and so on

Reply:
In a follow up refactor I'd like to make these types more convenient, in particular I'd like to extend the `QValue` interface. At that time it may make more sense how to divide the code

Reply:
moved Equals out, halving size of qvalue.go